### PR TITLE
Update responsetype

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @EPashkin  @sdroege

### CODE REVIEW

File: src/bin/gtktest.rs [hunk 1]
```diff
@@ -112,8 +112,12 @@ mod example {
...
             let entry = upgrade_weak!(entry_weak);
 
-            let dialog = Dialog::new_with_buttons(Some("Hello!"), Some(&window), gtk::DialogFlags::MODAL,
-                &[("No", 0), ("Yes", 1), ("Yes!", 2)]);
+            let dialog = Dialog::new_with_buttons(Some("Hello!"),
+                                                  Some(&window),
+                                                  gtk::DialogFlags::MODAL,
+                                                  &[("No", ResponseType::Other(0)),
+                                                    ("Yes", ResponseType::Other(1)),
+                                                    ("Yes!", ResponseType::Other(2))]);
```
Comment (EPashkin) [inline src/bin/gtktest.rs]: Maybe better declare global statics or constants?

**Comment by EPashkin** [comment]
Thanks again, 👍 even without constants

File: src/bin/gtktest.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/bin/gtktest.rs]: Didn't think about it but I could yes. It just seems a bit too much something used only once. What do you think @sdroege?

File: src/bin/listbox_model.rs [hunk 2]
```diff
@@ -121,8 +123,8 @@ fn build_ui(application: &gtk::Application) {
             let window = upgrade_weak!(window_weak);
 
             let dialog = gtk::Dialog::new_with_buttons(Some("Edit Item"), Some(&window), gtk::DialogFlags::MODAL,
-                &[("Close", 0)]);
-            dialog.set_default_response(0);
+                &[("Close", ResponseType::Other(0))]);
+            dialog.set_default_response(ResponseType::Other(0));
```
Comment (sdroege) [inline src/bin/listbox_model.rs]: `ReponseType::Close` exists, so why not use that? :)

File: src/bin/listbox_model.rs [hunk 3]
```diff
@@ -141,7 +143,7 @@ fn build_ui(application: &gtk::Application) {
             let dialog_weak = dialog.downgrade();
             entry.connect_activate(move |_| {
                 let dialog = upgrade_weak!(dialog_weak);
-                dialog.response(0);
+                dialog.response(ResponseType::Other(0));
```
Comment (sdroege) [inline src/bin/listbox_model.rs]: `ResponseType::Close`

File: src/bin/listbox_model.rs [hunk 4]
```diff
@@ -185,16 +187,16 @@ fn build_ui(application: &gtk::Application) {
             let window = upgrade_weak!(window_weak);
 
             let dialog = gtk::Dialog::new_with_buttons(Some("Add Item"), Some(&window), gtk::DialogFlags::MODAL,
-                &[("Ok", 0), ("Cancel", 1)]);
-            dialog.set_default_response(0);
+                &[("Ok", ResponseType::Other(0)), ("Cancel", ResponseType::Other(1))]);
```
Comment (sdroege) [inline src/bin/listbox_model.rs]: `ResponseType::Ok`, `::Cancel`

File: src/bin/listbox_model.rs [hunk 5]
```diff
@@ -185,16 +187,16 @@ fn build_ui(application: &gtk::Application) {
...
+            dialog.set_default_response(ResponseType::Other(0));
```
Comment (sdroege) [inline src/bin/listbox_model.rs]: `ResponseType::Ok`

File: src/bin/listbox_model.rs [hunk 6]
```diff
@@ -185,16 +187,16 @@ fn build_ui(application: &gtk::Application) {
...
 
             let content_area = dialog.get_content_area();
 
             let entry = gtk::Entry::new();
             let dialog_weak = dialog.downgrade();
             entry.connect_activate(move |_| {
                 let dialog = upgrade_weak!(dialog_weak);
-                dialog.response(0);
+                dialog.response(ResponseType::Other(0));
```
Comment (sdroege) [inline src/bin/listbox_model.rs]: `ResponseType::Ok`

File: src/bin/listbox_model.rs [hunk 7]
```diff
@@ -203,7 +205,7 @@ fn build_ui(application: &gtk::Application) {
 
             dialog.connect_response(clone!(model, entry, spin_button => move |dialog, resp| {
                 if let Some(text) = entry.get_text() {
-                    if !text.is_empty() && resp == 0 {
+                    if !text.is_empty() && resp == ResponseType::Other(0) {
```
Comment (sdroege) [inline src/bin/listbox_model.rs]: `ResponseType::Ok`

File: src/bin/gtktest.rs (hunk 1 above)
Comment (sdroege) [reply src/bin/gtktest.rs]: There are `ResponseType::Yes` and `::No`, why not use those? :) And for the third one, maybe just call it `"Custom"` and use `::Other(0)` or so for that?

**Comment by GuillaumeGomez** [comment]
@sdroege It was so obvious... I feel dumb for not thinking about. Thanks a lot for catching it!

File: src/bin/listbox_model.rs [hunk 8]
```diff
@@ -141,7 +143,7 @@ fn build_ui(application: &gtk::Application) {
             let dialog_weak = dialog.downgrade();
             entry.connect_activate(move |_| {
                 let dialog = upgrade_weak!(dialog_weak);
-                dialog.response(0);
+                dialog.response(ResponseType::Close);
```
Comment (EPashkin) [inline src/bin/listbox_model.rs]: Still 0 here "send response 0 to the dialog"

Comment (EPashkin) [reply src/bin/listbox_model.rs]: I meant comment above

**Pull request merged**